### PR TITLE
0.12 rename `edge` tag to `edge-acorn`

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by vvagaytsev** [comment]
Keeping this a draft for now. To avoid accidental premature merge.

**Review by stefreak — DISMISSED** [review]

File: scripts/release.ts [hunk 1]
```diff
@@ -326,8 +326,8 @@ async function stripPrereleaseTags(tags: string[], version: string) {
     }
   }
 
-  // We also need to remove the "edge" tag
-  await execa("git", ["tag", "-d", "edge"])
+  // We also need to remove the "edge-acorn" tag
+  await execa("git", ["tag", "-d", "edge-acorn"])
```
Comment (stefreak) [inline scripts/release.ts:330]: Might make sense to just delete all kinds of edge tags
```suggestion
  await execa("git", ["tag", "-d", "edge"])
  await execa("git", ["tag", "-d", "edge-acorn"])
  await execa("git", ["tag", "-d", "edge-bonsai"])
```

Comment (vvagaytsev) [reply scripts/release.ts:330]: Good catch, thanks! I'm not sure about the old `edge` tag. We can do it manually only once. I'm not sure if `execa` will raise an exception if the tag is not found. I'll check that.

Comment (vvagaytsev) [reply scripts/release.ts:330]: Actually, `edge-bonsai` should not be deleted here. It belongs to a separate 0.13 release process and must be deleted in the main branch here: https://github.com/garden-io/garden/blob/main/scripts/release.ts#L330

I'll create a separate PR to fix that.

Comment (vvagaytsev) [reply scripts/release.ts]: Added a [helper function](https://github.com/garden-io/garden/pull/4521/commits/2f813f09b4ea07feb09bb37fdbd920527ec017b3) for fail-safe tags removal, and [removed](https://github.com/garden-io/garden/pull/4521/commits/17b8736b359cad6357a00731554f57e1b36c68d0) both `edge` and `edge-acorn`.

Comment (stefreak) [reply scripts/release.ts]: Awesome

**Pull request merged**